ee/ucp/interlock/index.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Layer 7 routing overview
+description: Learn how to route layer 7 traffic to your Swarm services
+keywords: routing, UCP, interlock, load balancing
+---
+
+Application-layer (Layer 7) routing is the application routing and load balancing (ingress routing) system included with Docker Enterprise for Swarm orchestration. Interlock architecture takes advantage of the underlying Swarm components to provide scalable Layer 7 routing and Layer 4 VIP mode functionality.
+
+Interlock is specific to the Swarm orchestrator. If you're trying to route
+traffic to your Kubernetes applications, check
+[layer 7 routing with Kubernetes.](../kubernetes/layer-7-routing.md)
+
+Interlock uses the Docker Remote API to automatically configure extensions such as NGINX or HAProxy for application traffic. Interlock is designed for:
+
+- Full integration with Docker (Swarm, Services, Secrets, Configs)
+- Enhanced configuration (context roots, TLS, zero downtime deploy, rollback)
+- Support for external load balancers (nginx, haproxy, F5, etc) via extensions
+- Least privilege for extensions (no Docker API access)
```

Comment:
```suggestion
- Least privilege proxy reconfiguration via an extension interface with no Docker API access
```

- explain what an extension is since I removed the previous line